src/Npgsql/TypeHandlers/NetworkHandlers/InetHandler.cs

```diff
@@ -62,6 +63,20 @@ NpgsqlInet INpgsqlSimpleTypeHandler<NpgsqlInet>.Read(NpgsqlReadBuffer buf, int l
 
         #region Write
 
+        protected internal override int ValidateObjectAndGetLength(object value, ref NpgsqlLengthCache lengthCache, NpgsqlParameter parameter)
+            => value == null || value is DBNull
+                ? -1
+                : value is IPAddress ip
+                    ? ValidateAndGetLength(ip, parameter)
+                    : throw new InvalidCastException($"Can't write CLR type {value.GetType().Name} to database type {PgDisplayName}");
+
+        protected internal override Task WriteObjectWithLength(object value, NpgsqlWriteBuffer buf, NpgsqlLengthCache lengthCache, NpgsqlParameter parameter, bool async)
+            => value == null || value is DBNull // For null just go through the default WriteWithLengthInternal
+                ? WriteWithLengthInternal<DBNull>(null, buf, lengthCache, parameter, async)
+                : value is IPAddress ip
+                    ? WriteWithLengthInternal(ip, buf, lengthCache, parameter, async)
+                    : throw new InvalidCastException($"Can't write CLR type {value.GetType().Name} to database type {PgDisplayName}");
```

Comment:
I think this handler is reused by cidr too, which might explain the failing tests.

Reply:
Makes sense... I'll look into it again tomorrow, we may need to split into two handlers...